github_helper.py

```diff
@@ -0,0 +1,159 @@
+"""
+github_helper provides several methods to help interface with Github
+"""
+
+import requests
+import urlparse
+
+GITHUB_ORG = 'apache'
+SECRET_FILE = '../../github_auth/apache-beam.secret'
+SOURCE_REMOTE = 'github'
+TARGET_BRANCH = 'master'
+TARGET_REMOTE = 'apache'
+
+GITHUB_API_ROOT = 'https://api.github.com'
+GITHUB_REPO_FMT_URL = GITHUB_API_ROOT + '/repos/{0}/{1}'
+GITHUB_COMMENTS_ENDPOINT = '/{}/comments'
+GITHUB_ISSUES_ENDPOINT = '/issues/'
+GITHUB_PULLS_ENDPOINT = '/pulls/'
+
+BOT_NAME = 'merge-bot'
+GITHUB_SECRET = '../../github_auth/mergebot.secret'
+
+
+class GithubHelper(object):
+    """
+    GithubHelper provides a lightweight authenticated wrapper for the Github
+    API.
+    """
+
+    def __init__(self, org, repo):
+        self.org = org
+        self.repo = repo
+        with open(GITHUB_SECRET, 'r') as key_file:
+            self.bot_key = key_file.read().strip()
+        print('Loaded key file.')
+
+    def fetch_prs(self):
+        """ Gets the PRs for the configured repository
+
+        Returns:
+            An iterable containing instances of GithubPR if successful.
+            None otherwise.
```

Comment:
Why not an empty list otherwise? Are you using the distinction between "failure to get PRs" (None) and "no PRs" (empty list)? If so, consider raising an exception to represent the exceptional case. Otherwise, empty list seems reasonable.